### ISSUE TITLE
Install/1st run bug?

### ISSUE THREAD

**Issue by annakirkpatrick** [OptiKey/OptiKey#700]
I've been helping a friend get started using OptiKey with an eye tracker and external switch. She is getting a nonspecific "OptiKey tripped over its feet" error when trying to launch OptiKey Pro. 

Details:

Windows 10, OptiKey Pro v3.0.5

She first installed OptiKey Pro a few weeks ago. This was right after I told her about OptiKey, and she wanted to give it a try with the mouse to see how it worked before investing in an eye tracker. At this point, all was working fine.

A couple days ago, she got her eyetracker in the mail. It's the new Tobii eye tracker 5. She got the drivers from Toby installed, and the gaze trace verifies that the eye tracker is functional.

She was able to change the input source from mouse to eyetracker (she used the tobii 4C option), and, aside from some accuracy issues, this worked. At this point, she was using the dwell selection method.

She then started experimenting with settings, including changing the selection method from dwell to keyboard key.

I got on Zoom with her today so that I could share her screen and help her debug some of the accuracy issues. When she attempted to run OptiKey, she got the nonspecific "OptiKey tripped over its feet" error. (Additionally, the space for OptiKey was allocated at the top of the screen and remained empty.)

She rebooted the computer, and had the exact same issue. I then instructed her to uninstall OptiKey and then reinstall it. Same issue.

This is really weird because her first install of OptiKey Pro was successful. As far as I can tell the only difference is that she selected the eye tracker input method in the installation wizard (as opposed to the mouse, which she selected the first time). And the fact that the tobii service was running at the time of install. Of course it's possible that some user settings were not removed by the uninstaller, as well.

I was able to get a log file from the first time she tried to launch OptiKey after the reinstall. It is uploaded here. In particular, note the exception at line 302.

[Log.txt](https://github.com/OptiKey/OptiKey/files/4997088/Log.txt)

Any ideas what might be going on here?

**Comment by kmcnaught** [comment]
This looks suspiciously similar to this bug: https://github.com/OptiKey/OptiKey/issues/679, which *should* be fixed in 3.0.6 (currently pre-release) but I'm aware that a few similar issues have come up that might only have been fixed piecemeal, so the underlying issue might still be lurking.

Can you suggest she tries the new pre-release and report back any further issues with key selection.

**Comment by annakirkpatrick** [comment]
@kmcnaught Thanks for the response. My friend was able to get an older version of OptiKey running. I had her try version 2.32.0 (the last stable release before OptiKey was split into 4 separate programs), and it worked. So she's good.

But I will run some installs later today and see if I can reproduce the issue and see if it resolves with the latest prerelease. I'll let you know.

**Comment by JuliusSweetland** [comment]
@annakirkpatrick @kmcnaught Kirsty beat me to it! She should try 3.0.6 (currently here: https://github.com/OptiKey/OptiKey/releases/tag/v3.0.6_pre-release_5) - that should fix it.

If anyone has issues related to the settings then uninstalling/reinstalling will NOT work, as user settings are retained across restarts and re-installs. To clear everything back to factory settings you need to do this: https://github.com/Optikey/Optikey/wiki/Factory-reset-all-settings

**Comment by annakirkpatrick** [comment]
@JuliusSweetland Thanks for the clarity on user settings and the fact that they are preserved across uninstall/reinstall. I actually did find the page you referenced after I wrote my initial post. I had my friend try clearing the user settings, and that didn't solve the problem.

@kmcnaught @JuliusSweetland I was able to duplicate the error she experienced. (I hadn't upgraded to version 3 until today.) The error did appear right after I changed the selection method to be a keyboard key. I installed the prerelease 3.0.6, and all seems to be good. I'm actually using it right now. So I think this one is solved.
